regimes/mx/food_vouchers_complement.go

```diff
@@ -0,0 +1,122 @@
+package mx
+
+import (
+	"regexp"
+
+	"github.com/invopop/gobl/cal"
+	"github.com/invopop/gobl/cbc"
+	"github.com/invopop/gobl/num"
+	"github.com/invopop/validation"
+)
+
+// Constants for the precision of the complement's amounts
+const (
+	FoodVouchersFinalPrecision = 2
+)
+
+// Complement's Codes Patterns
+const (
+	CURPPattern           = "^[A-Z][A,E,I,O,U,X][A-Z]{2}[0-9]{2}[0-1][0-9][0-3][0-9][M,H][A-Z]{2}[B,C,D,F,G,H,J,K,L,M,N,Ñ,P,Q,R,S,T,V,W,X,Y,Z]{3}[0-9,A-Z][0-9]$"
+	SocialSecurityPattern = "^[0-9]{11}$"
+)
+
+// Complement's Codes Regexps
+var (
+	CURPRegexp           = regexp.MustCompile(CURPPattern)
+	SocialSecurityRegexp = regexp.MustCompile(SocialSecurityPattern)
+)
+
+// FoodVouchersComplement carries the data to produce a CFDI's "Complemento de
+// Vales de Despensa" (version 1.0) providing detailed information about food
+// vouchers issued by an e-wallet supplier to its customer's employees.
+//
+// This struct maps to the `ValesDeDespensa` root node in the CFDI's complement.
+type FoodVouchersComplement struct {
+	// Customer's employer registration number (maps to `registroPatronal`).
+	EmployerRegistration string `json:"employer_registration,omitempty" jsonschema:"title=Employer Registration"`
+	// Customer's account number (maps to `numeroDeCuenta`).
+	AccountNumber string `json:"account_number" jsonschema:"title=Account Number"`
+	// Sum of all line amounts (calculated, maps to `total`).
+	Total num.Amount `json:"total" jsonschema:"title=Total" jsonschema_extras:"calculated=true"`
+	// List of food vouchers issued to the customer's employees (maps to `Conceptos`).
+	Lines []*FoodVouchersLine `json:"lines" jsonschema:"title=Lines"`
+}
+
+// FoodVouchersLine represents a single food voucher issued to the e-wallet of
+// one of the customer's employees. It maps to one `Concepto` node in the CFDI's
+// complement.
+type FoodVouchersLine struct {
+	// Identifier of the e-wallet that received the food voucher (maps to `Identificador`).
+	EWalletID cbc.Code `json:"e_wallet_id" jsonschema:"title=E-wallet Identifier"`
+	// Date and time of the food voucher's issue (maps to `Fecha`).
+	IssueDateTime cal.DateTime `json:"issue_date_time" jsonschema:"title=Issue Date and Time"`
+	// Employee's tax identity code (maps to `rfc`).
+	EmployeeTaxCode cbc.Code `json:"employee_tax_code" jsonschema:"title=Employee's Tax Identity Code"`
+	// Employee's CURP ("Clave Única de Registro de Población", maps to `curp`).
+	EmployeeCURP cbc.Code `json:"employee_curp" jsonschema:"title=Employee's CURP"`
+	// Employee's name (maps to `nombre`).
+	EmployeeName string `json:"employee_name" jsonschema:"title=Employee's Name"`
+	// Employee's Social Security Number (maps to `numSeguridadSocial`).
+	EmployeeSocialSecurity cbc.Code `json:"employee_social_security,omitempty" jsonschema:"title=Employee's Social Security Number"`
```

Comment:
Whenever I see namespaces like this I think they could be inside their own object. Would a `FoodVoucherEmployee` object make sense?

Reply:
I think so... we're renaming everything anyway from Spanish to English, so I do think its reasonable.

Reply:
Cool! Based on [previous feedback](https://github.com/invopop/gobl/pull/192#discussion_r1315048006), I was following the general rule of "stick[ing] as closely as possible to the original source structure, with translated fields". But, I agree that this is a minor deviation from that general rule that makes things nicer for the user. Addressed in 90eaae82147fe06b95f7322ecfb563971bf50cb5.